pkg/server/middleware/request_limits.go

```diff
@@ -2,21 +2,32 @@ package middleware
 
 import (
 	"bytes"
+	"context"
 	"fmt"
 	"io"
+	"net"
 	"net/http"
 
+	"github.com/pkg/errors"
+
+	"github.com/go-kit/log"
 	"github.com/go-kit/log/level"
 	"github.com/grafana/mimir/pkg/util/spanlogger"
 )
 
+const (
+	StatusClientClosedRequest = 499
```

Comment:
I checked this repo and found that 499 is already used in pkg/errorx/http_translator.go
https://github.com/grafana/mimir-proxies/blob/6b2a4f93edfe67b44217edb1d6cf6cac196b3890/pkg/errorx/http_translator.go#L13
and in pkg/datadog/apiwrite/apiwrite.go
https://github.com/grafana/mimir-proxies/blob/6b2a4f93edfe67b44217edb1d6cf6cac196b3890/pkg/datadog/apiwrite/apiwrite.go#L26

So maybe we can keep 499 here as well?

Reply:
Nice find @mar4uk ! I think we have an issue where we point out that we need mimir-proxies to behave more like otlp-gateway, so keeping the 499 does exactly that - I think we're all agreed this is the way to standardise (499, instead of a 408 that's 50/50 a 4xx/5xx, etc).

Reply:
https://github.com/grafana/app-o11y-kwl-endpoint/issues/149#issue-1701072635

Reply:
(issue mentioning we need a 499 instead of 500 response, to mirror otlp-gateway)